python/pacemaker/_cts/validate.py

```diff
@@ -0,0 +1,62 @@
+"""A module providing XML validation utilities."""
+
+__copyright__ = "Copyright 2024 the Pacemaker project contributors"
+__license__ = "GNU General Public License version 2 or later (GPLv2+)"
+
+__all__ = ["validate"]
+
+import os
+import subprocess
+import sys
+
+from pacemaker._cts.errors import XmlValidationError
+from pacemaker._cts.process import pipe_communicate
+from pacemaker.buildoptions import BuildOptions
+
+
+def find_validator(rng_file):
+    """
+    Return the command line used to validate XML output.
+
+    If no validator is found, return None.
+    """
+    if os.access(BuildOptions.XMLLINT_PATH, os.X_OK):
+        if rng_file is None:
+            return ["xmllint", "-"]
+
+        return [BuildOptions.XMLLINT_PATH, "--relaxng", rng_file, "-"]
+
+    raise FileNotFoundError("Could not find validator for %s" % rng_file)
+
+
+def rng_directory():
+    """Return the directory containing RNG schema files."""
+    if os.environ.get("PCMK_schema_directory", "") != "":
+        return os.environ["PCMK_schema_directory"]
+
+    for path in sys.path:
+        if os.path.exists("%s/cts-fencing.in" % path):
+            return "xml"
```

Comment:
how does this work? wouldn't xml always be relative to the cwd?

Reply:
Yeah, I think the only way this makes sense is if it's `return "%s/xml" % path`.  Fixed.